File: tests/test_lc_125_valid_palindrome.c
#include <unity.h>
#include "lc_125_valid_palindrome.c"

#define TRUE    1
#define FALSE   0

void setup()
{
    //
}

void teardown()
{
    //
}

void test_isPalindrome()
{
    //Test 1
    char s[]    = "A man, a plan, a canal: Panama";
    int output  = TRUE;

    TEST_ASSERT (isPalindrome(s) == output);

}

File: tests/test_lc_125_valid_palindrome.py
import setup_paths

from lc_125_valid_palindrome import ValidPalindrome

def test_solve():

    #Test 1
    s       = "A man, a plan, a canal: Panama"
    output  = True

    solution = ValidPalindrome(s)

    assert solution.solve() == output

    #Test 2
    s       = "race a car"
    output  = False

    solution = ValidPalindrome(s)

    assert solution.solve() == output

    #Test 3
    s       = " "
    output  = True

    solution = ValidPalindrome(s)

    assert solution.solve() == output